components/image/image.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright Lingzhu Ltd. 2020 All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package image
+
+import (
+	"image"
+	"image/color"
+	"image/draw"
+	"image/png"
+	"io/ioutil"
+	"os"
+
+	"github.com/csiabb/donation-service/common/log"
+	"github.com/csiabb/donation-service/common/rest"
+	"github.com/csiabb/donation-service/components/wx"
+
+	"github.com/golang/freetype"
+	"github.com/golang/freetype/truetype"
+	"golang.org/x/image/math/fixed"
+)
+
+var (
+	logger = log.MustGetLogger("components/image")
+)
+
+// Client image services client
+type Client struct {
+	ImageConfig *Config
+	Bg          *image.NRGBA
+	Font        *freetype.Context
+	FontType    *truetype.Font
+}
+
+// Init initializes a new background image
+func (c *Client) Init() error {
+	imgFile, err := os.Open(c.ImageConfig.BackgroundPath)
+	if err != nil {
+		logger.Errorf("failed to read bg path %s: %s", c.ImageConfig.BackgroundPath, err)
+		return err
+	}
+	defer imgFile.Close()
+
+	pngImg, err := png.Decode(imgFile)
+	if err != nil {
+		logger.Errorf("failed to bg decode : %s", err)
+		return err
+	}
+
+	bg := image.NewNRGBA(pngImg.Bounds())
+	for y := 0; y < bg.Bounds().Dy(); y++ {
+		for x := 0; x < bg.Bounds().Dx(); x++ {
+			bg.Set(x, y, pngImg.At(x, y))
+		}
+	}
+	c.Bg = bg
+
+	fontBytes, err := ioutil.ReadFile(c.ImageConfig.FontPath)
+	if err != nil {
+		logger.Errorf("failed to read font path %s: %s", c.ImageConfig.FontPath, err)
+		return err
+	}
+
+	fontType, err := freetype.ParseFont(fontBytes)
+	if err != nil {
+		logger.Errorf("failed to parse font : %s", err)
+		return err
+	}
+	c.FontType = fontType
+
+	font := freetype.NewContext()
+	font.SetDPI(72)
+	font.SetFont(fontType)
+	font.SetClip(bg.Bounds())
+	font.SetDst(bg)
+
+	c.Font = font
+
+	return nil
+}
+
+// CreateQrCode  create a wx qr code
+func (c *Client) CreateWXQrCode(appID string, secret string, scene string) (img image.Image, err error) {
```

Comment:
build failed:
comment on exported method Client.CreateWXQrCode should be of the form "CreateWXQrCode ...